src/unix/osd.c

```diff
@@ -65,16 +67,29 @@ int fi_fd_nonblock(int fd)
 	return 0;
 }
 
-int fi_wait_cond(pthread_cond_t *cond, pthread_mutex_t *mut, int timeout)
+/* timeout is expressed in miliseconds */
+int fi_wait_cond(pthread_cond_t *cond, pthread_mutex_t *mut, int timeout_ms)
 {
+	int ret;
+	struct timeval tv;
+	uint64_t t;
 	struct timespec ts;
 
-	if (timeout < 0)
+	if (timeout_ms < 0)
 		return pthread_cond_wait(cond, mut);
 
-	clock_gettime(CLOCK_REALTIME, &ts);
-	ts.tv_sec += timeout / 1000;
-	ts.tv_nsec += (timeout % 1000) * 1000000;
+	ret = gettimeofday(&tv, NULL);
+	if (ret < 0)
+		return ret;
+
+	/* Set t to now, expressed in microseconds */
+	t = (tv.tv_sec + (timeout_ms / 1000)) * 1000000;
+	t += (timeout_ms % 1000) * 1000;
+	t += tv.tv_usec;
```

Comment:
Btw, we have helper calls fi_gettime_ms, fi_gettime_us, which may help simplify the code here.  Those just call gettimeofday internally.